src/contributing.md

```diff
@@ -0,0 +1,140 @@
+  - If the content is not available, consider having placeholders using the estimated size (delegates, dummy items)
+
+## QML well-behaved components checklist
+
+### Sizing support
+
+- Define size hints appropriately. Define implicit properties (`impicitWidth` and `implicitHeight`)
+  - They are used to break polishing binding loops for adaptive control
+  - All the containers use the implicit size for deriving the initial size (Layouts, Popups, Delegates, GridView, ListView ...). Size hints combined with resize adaptability are the appropriate way to have reactive controls.
+  - For complex controls, look if layouts are the choice as the main position driver. [Layouts](###Layouts)
+
+  ```qml
+  Item {
+    id: root
+    implicitWidth: mainLayout.implicitWidth
+    implicitHeight: mainLayout.implicitHeight
+    RowLayout {	// Column, Grid
+       id: mainLayout
+       // ...
+    }
+  ```
+
+- Define a default size to work with positioners (`Row`, `Column`, `Grid`). If appropriate, source them from implicit dimensions and expect the owner to overwrite them.
+  - Don't mix hints with sizes, it will create binding loops
+- Adapt to the requested size.
+Have the base control as the component's root. This way, control inherits all the interface and reduce the code.
+If it doesn't map to an existing one, use `Item` (never use layout or positioners as base control, they don't behave as expected)
+Have public properties at the top and make them explicit
+
+## Topics
+
+### Layouts
+
+Hierarchically sources hints from children. Implicit properties will have the recommended aggregated size.
+
+Layouts as children of other layouts have `fillWidth` and `fillHeight` enabled; controls don't.
+
+Use `Layout.prefferedWidth` and `Layout.prefferedHeight` attached properties to overwrite item's implicitWidth if they are not appropriate to the design.
+
+### Scope
+
+Component 'id's are not accessible outside component's scope
+
+- Example
+
+  ```qml
+  Item {
+    // Probably won't work as intended. If anoter `testComp` instan is defined in QML's document hierarchy model and has a `testProp` property, that will be sourced instead
+    property bool booleanProp: testComp.testProp
+    // Same behaviour if the TestComponent is defined in a file TestComponent.qml
+    component TestComponent: Item {
+      id: testComp
+      property bool testProp: false
+    }
+  }
+  ```
+
+- If in doubt, explicitly use an instance variable to access properties
+- If the scope is clear and there is no ambiguity, use directly the property for readability
+
+## Testing
+
+Test in isolation
+
+- Use `qmlproject`s for quick complex scenarios
+- Use [`QtQuick.TestCase`](https://doc.qt.io/qt-5/qtquicktest-index.html)
+
+Integration tests
+
+- Use sandbox test app
+- Use QML `qmlproject`s for quick debugging with layouts
+
+Try scenarios
+
+- Embed in Layouts with different properties: fixed size, min-max, not size set
```

Comment:
I like this approach, maybe we should make it a part of the sandbox test app? We can even try to generalize it, so that the required code would be:
```
TestItemInLayout {
   componentSrc: "TestControl.qml"
}
```

Reply:
Great idea, let's bring this experiment.